src/main/java/de/tum/in/www1/artemis/domain/ProgrammingExercise.java

```diff
@@ -572,10 +577,23 @@ public boolean areManualResultsAllowed() {
         return getAssessmentType() == AssessmentType.SEMI_AUTOMATIC && (relevantDueDate == null || relevantDueDate.isBefore(ZonedDateTime.now()));
     }
 
+    /**
+     * This checks if the current result is rated and has a completion date.  
+     * @param result The current result
+     * @return true if the result is manual and assessed, false otherwise
+     */
     private boolean checkForRatedAndAssessedResult(Result result) {
+        return Boolean.TRUE.equals(result.isRated()) && checkForAssessedResult(result);
+    }
+
+    /**
+     * This checks if the current result has a completion date and if the assessment is over
+     * @param result The current result
+     * @return true if the result is manual and the assessment is over or it is an automatic result, false otherwise
+     */
+    private boolean checkForAssessedResult(Result result) {
         boolean isAssessmentOver = getAssessmentDueDate() == null || getAssessmentDueDate().isBefore(ZonedDateTime.now());
-        return Boolean.TRUE.equals(result.isRated()) && result.getCompletionDate() != null
-                && ((result.isManual() && isAssessmentOver) || result.getAssessmentType().equals(AssessmentType.AUTOMATIC));
+        return result.getCompletionDate() != null && ((result.isManual() && isAssessmentOver) || result.getAssessmentType().equals(AssessmentType.AUTOMATIC));
```

Comment:
```suggestion
        return result.getCompletionDate() != null && (result.isManual() && isAssessmentOver || result.getAssessmentType().equals(AssessmentType.AUTOMATIC));
```

Reply:
&& has stronger binding than ||

Reply:
yeah you are right, but I do not think this is a big issue. With the brackets it is immediately clear what's going on. :)

Reply:
i also like the notation with brackets in longer statements, because it is less confusing in my opinion.